scanner/datastore/postgres/vuln_update.go

```diff
@@ -3,48 +3,81 @@ package postgres
 import (
 	"context"
 	"errors"
-	"fmt"
-	"net/http"
-	"strings"
 	"time"
 
 	"github.com/jackc/pgx/v4"
 )
 
-const lastVulnUpdateKey = `last-vuln-update`
-
-// GetLastVulnerabilityUpdate retrieves the last vulnerability update from the database.
+// GetLastVulnerabilityUpdate implements MatcherMetadataStore.GetLastVulnerabilityUpdate.
 //
-// The returned time will be in the form of http.TimeFormat.
+// Assumes the last update is the oldest update timestamp in the table.
 func (m *matcherMetadataStore) GetLastVulnerabilityUpdate(ctx context.Context) (time.Time, error) {
-	const selectTimestamp = `SELECT timestamp FROM last_vuln_update WHERE key = $1`
-
-	var t string
-	row := m.pool.QueryRow(ctx, selectTimestamp, lastVulnUpdateKey)
+	const selectTimestamp = `
+		SELECT update_timestamp
+		FROM last_vuln_update
+		WHERE update_timestamp IS NOT NULL ORDER BY update_timestamp LIMIT 1;`
+	row := m.pool.QueryRow(ctx, selectTimestamp)
+	var t time.Time
 	err := row.Scan(&t)
 	if errors.Is(err, pgx.ErrNoRows) {
 		return time.Time{}, nil
 	}
 	if err != nil {
 		return time.Time{}, err
 	}
+	return t.UTC(), nil
+}
 
-	timestamp, err := time.Parse(http.TimeFormat, strings.TrimSpace(t))
+// SetLastVulnerabilityUpdate implements MatcherMetadataStore.SetLastVulnerabilityUpdate.
+//
+// We use one row for each updater, keyed by their name.
+func (m *matcherMetadataStore) SetLastVulnerabilityUpdate(ctx context.Context, timestamp time.Time, updaterName string) error {
+	const insertTimestamp = `
+		INSERT INTO last_vuln_update (key, update_timestamp)
+		VALUES ($1, $2)
+		ON CONFLICT (key) DO UPDATE SET update_timestamp = $2;`
+	_, err := m.pool.Exec(ctx, insertTimestamp, updaterName, timestamp.UTC().Truncate(time.Second))
```

Comment:
Few questions on this (mostly just for my own info):
1. Do you know why we did `timestamp.Format(http.TimeFormat)` before?
2. Why are we changing it?
3. Why do we need to truncate the seconds?

Reply:
> Do you know why we did timestamp.Format(http.TimeFormat) before?

No.

> Why are we changing it?

We are not changing it, specifically. We are creating a new column. But the new column is not a string, so we can `ORDER BY update_timestamp`.

> Why do we need to truncate the seconds?

To match timestamps within seconds apart, it is very unlikely updates will happen within less than seconds. This simplifies tests, readability of logs, etc.